src/main/java/taskalyn/TaskManager.java

```diff
@@ -0,0 +1,129 @@
+            ui.printLines("Awesome bro! One task gone :D\n" +
+                    "      " + task.toString() + "\n" + "    Wah bro... " + getTaskSize() + (getTaskSize() > 1 ? " tasks already!" : " task already!"));
+        }
+        updateDatabase();
+    }
+
+    public int getTaskSize() {
+        return tasks.size();
+    }
+
+    public void listTasks() {
+        String taskString = "Here are the tasks in your list:\n" + "    ";
+
+        if (!tasks.isEmpty()) {
+            for (int i=0; i<tasks.size(); i++) {
+                if (i != tasks.size() - 1) {
+                    taskString += i + 1 + "." + tasks.get(i).toString() + "\n    ";
+                }
+                else {
+                    taskString += i + 1 + "." + tasks.get(i).toString();
+                }
+            }
+        }
+        else {
+            taskString += "Nothing!";
+        }
+
+        ui.printLines(taskString);
+    }
+
+    public void completeTask(int taskId) {
+        if (taskId <= tasks.size() + 1) {
+            tasks.get(taskId - 1).setComplete();
+            ui.printLines("Nice, I've marked this task as complete:\n" +
+                    "       " + tasks.get(taskId - 1).toString());
+        }
+        updateDatabase();
+    }
+
+    public void incompleteTask(int taskId) {
+        if (taskId <= tasks.size() + 1) {
+            tasks.get(taskId - 1).setIncomplete();
+            ui.printLines("Ok, I've marked this task as incomplete:\n" +
+                    "       " + tasks.get(taskId - 1).toString());
+        }
+        updateDatabase();
+    }
+
+    private void loadDatabase() {
+        try {
+            List<String> txtLines = this.database.readFromDatabase();
+            for (String line : txtLines) {
+                Task task = findTask(line);
+                tasks.add(task);
+            }
+        } catch (IOException e) {
+            System.out.println("Error reading database: " + e.getMessage());
+        }
+    }
+
+    private Task findTask(String line) {
+        String[] parts = line.split(" \\| ");
+        if (parts.length >= 3) {
+            String taskType = parts[0];
+            boolean isCompleted = parts[1].equals("1");
+            String taskInfo = parts[2];
+
+            switch (taskType) {
+            case "T":
+                return new TodoTask(taskInfo, isCompleted);
+
+            case "D":
+                if (parts.length == 4) {
+                    String date = parts[3];
+                    return new DeadlineTask(taskInfo, date, isCompleted);
+                }
+                break;
+
+            case "E":
+                if (parts.length == 5) {
+                    String toDate = parts[3];
+                    String fromDate = parts[4];
+                    return new EventTask(taskInfo, toDate, fromDate, isCompleted);
+                }
+                break;
+
+            default:
+            }
+        }
+        return null;
+    }
+
+    private void updateDatabase() {
```

Comment:
Perhaps you could add JavaDocs for this private method too as it is crucial in this class

Reply:
That's true. I'll add them.